app/src/main/java/com/android/sample/model/recipe/CreateRecipeViewModel.kt

```diff
@@ -0,0 +1,88 @@
+package com.android.sample.model.recipe
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.google.firebase.firestore.FirebaseFirestore
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+/**
+ * ViewModel for creating a new recipe.
+ *
+ * @property repository The repository used to add the recipe to Firestore.
+ */
+class CreateRecipeViewModel(private val repository: FirestoreRecipesRepository) : ViewModel() {
+
+  val recipeBuilder = RecipeBuilder()
+  private val _publishError = MutableStateFlow<String?>(null)
+  val publishError: StateFlow<String?>
+    get() = _publishError
+
+  fun updateRecipeName(name: String) {
+    recipeBuilder.setName(name)
+  }
+
+  fun updateRecipeCategory(category: String) {
+    recipeBuilder.setCategory(category)
+  }
+
+  fun updateRecipeArea(area: String) {
+    recipeBuilder.setArea(area)
+  }
+
+  fun updateRecipeInstructions(instructions: String) {
+    recipeBuilder.setInstructions(instructions)
+  }
+
+  fun updateRecipeThumbnail(url: String) {
+    recipeBuilder.setPictureID(url)
+  }
+
+  fun updateRecipeTime(time: String) {
+    recipeBuilder.setTime(time)
+  }
+
+  fun updateRecipeDifficulty(difficulty: String) {
+    recipeBuilder.setDifficulty(difficulty)
+  }
+
+  fun updateRecipePrice(price: String) {
+    recipeBuilder.setPrice(price)
+  }
+
+  fun addIngredient(ingredient: String, measurement: String) {
+    recipeBuilder.addIngredientAndMeasurement(ingredient, measurement)
+  }
```

Comment:
There is I think some needed check for non null `String`, I know that we have discussed that we will do it in the UI but I think it's great to maybe also put it here since we don't know if in a future feature, the person using this class will be aware that he needs to check for non null parameters in the UI.